lib/github_changelog_generator/generator/generator_fetcher.rb

```diff
@@ -40,8 +40,10 @@ def detect_actual_closed_dates(issues)
 
       issues.each_slice(MAX_THREAD_NUMBER) do |issues_slice|
         threads = []
-        issues_slice.each do |issue|
-          threads << Thread.new { find_closed_date_by_commit(issue) }
+        threads << Thread.new do
+          issues_slice.each do |issue|
+            find_closed_date_by_commit(issue)
+          end
         end
         threads.each(&:join)
```

Comment:
This is still wrong since the thread array and joining is inside of the `each_slice` loop. This means each thread is blocked by the join, but not the group of threads.

Reply:
But when I move the threading outside of the each_slice I get memory bloat again.

Reply:
Actually the original code wasn't so terrible. It just did 25 threads for the 25 issue slices, waiting for each slice. The problem seems to be threading at all because as I reduce the number of threads the memory usage *and* time to complete goes down. Removing threading makes it faster and takes less memory? Sure.